lib/hanami/routing/http_router.rb

```diff
@@ -158,10 +163,15 @@ def no_response(request, env)
       end
 
       # @api private
-      # @since 0.5.0
-      def rewrite_path_info(env, request)
-        super
-        env[SCRIPT_NAME] = @prefix.join(env[SCRIPT_NAME])
+      def rewrite_partial_path_info(env, request)
+        path_info_before = request.rack_request.path_info.dup
+        if request.path.empty?
+          env[PATH_INFO] = "/"
+          env[SCRIPT_NAME] += path_info_before
+        else
+          env[PATH_INFO] = "/#{URI.encode(request.path.join('/'))}"
+          env[SCRIPT_NAME] += path_info_before[0, path_info_before.size - env[PATH_INFO].size]
```

Comment:
@cyphactor Shouldn't we use `bytesize` instead of `size`? It's safer for Unicode chars.